monai/bundle/config_reader.py

```diff
@@ -24,123 +25,118 @@
         """
-        Export the config content to the specified file path.
-        Suppprt JSON and YAML formats.
+        Export the config content to the specified file path (currently support JSON and YAML files).
 
         Args:
             config: source config content to export.
-            filepath: target file path to save, supported postfixes: `.json`, `.yml`, `.yaml`.
-            kwargs: other arguments for `json.dump` or `yaml.safe_dump`, depends on file format.
-                for more details, please check:
-                https://docs.python.org/3/library/json.html#json.dump.
-                https://pyyaml.org/wiki/PyYAMLDocumentation.
+            filepath: target file path to save.
+            fmt: format of config content, currently support ``"json"`` and ``"yaml"``.
+            kwargs: other arguments for ``json.dump`` or ``yaml.safe_dump``, depends on the file format.
 
         """
         _filepath: str = str(Path(filepath))
+        writer = look_up_option(fmt.lower(), {"json", "yaml"})
         with open(_filepath, "w") as f:
-            if _filepath.lower().endswith(cls.suffixes[0]):
+            if writer == "json":
                 return json.dump(config, f, **kwargs)
-            if _filepath.lower().endswith(tuple(cls.suffixes[1:])):
+            if writer == "yaml":
                 return yaml.safe_dump(config, f, **kwargs)
-            raise ValueError("only support JSON or YAML config file so far.")
+            raise ValueError(f"only support JSON or YAML config file so far, got {writer}.")
 
     @classmethod
-    def extract_file_path(cls, src: str) -> Optional[Tuple[str, str]]:
+    def split_path_id(cls, src: str) -> Tuple[str, str]:
         """
-        extract a config file path from the source string, return path and the rest string.
-        return `None` if can't find any config file path.
+        Split `src` string into two parts: a config file path and component id.
+        The file path should end with `(json|yaml|yml)`. The component id should be separated by `#` if it exists.
 
         Args:
-            src: source string to extract, it can be a config file path with / without additional information.
-                for example: "/data/config.json", "/data/config.json#net#<args>".
+            src: source string to split.
 
         """
-        pattern = "|".join(cls.suffixes)
-        result = re.findall(pattern, src, re.IGNORECASE)
-        if len(result) != 1:
-            # src should only contain 1 file
-            return None
-        items = src.split(result[0])
-        # return file path and the rest
-        return items[0] + result[0], items[1]
+        path, *ids = f"{src}".rsplit(ReferenceResolver.sep, 1)
+        ids = ids[0] if ids else ""
+        path_name = cls.path_match.findall(path)
+        if not path_name:
+            return "", src  # the src is a pure id
+        if len(path_name) < 1 and len(path_name[0]) < 1:
```

Comment:
I don't think this `if` will hapen, maybe we can simplify the logic to?
```py
        if not path_name:
            return "", src  # the src is a pure id
        return path_name[0][0], ids
```

Thanks.

Reply:
It'll be here for some corner cases for 'src'

Reply:
Sure, updated in the latest commit.

Thanks.